base/dates/parse.jl

```diff
@@ -200,3 +235,38 @@ function Base.parse(::Type{DateTime}, s::AbstractString, df::typeof(ISODateTimeF
     @label error
     throw(ArgumentError("Invalid DateTime string"))
 end
+
+function Base.parse{T<:TimeType}(
+    ::Type{T}, str::AbstractString, df::DateFormat=default_format(T),
+)
+    nt = tryparse_internal(T, str, df, true)
+    T(unsafe_get(nt)...)
+end
+
+function Base.tryparse{T<:TimeType}(
+    ::Type{T}, str::AbstractString, df::DateFormat=default_format(T),
+)
+    nt = tryparse_internal(T, str, df, false)
+    if isnull(nt)
+        Nullable{T}()
+    else
+        Nullable{T}(T(unsafe_get(nt)...))
+    end
+end
+
+@generated function Base.parse(::Type{Vector}, str::AbstractString, df::DateFormat)
```

Comment:
We should consider having "Parser types" (kinda like the token types in this module) and using their objects of those types as the first argument to parse rather than have it always be `Type` of the return value. That is not sustainable as you can see.